runtime/connections.go

```diff
@@ -381,7 +347,51 @@ func (c *ConnectorConfig) setPreset(k, v string, force bool) {
 		return
 	}
 	if c.Preset == nil {
-		c.Preset = make(map[string]string)
+		c.Preset = make(map[string]any)
 	}
 	c.Preset[k] = v
 }
+
+func deepMerge(dst, src map[string]any) map[string]any {
+	for k, v := range src {
+		if vMap, ok := v.(map[string]any); ok {
+			if dstMap, ok := dst[k].(map[string]any); ok {
+				dst[k] = deepMerge(dstMap, vMap)
+				continue
+			}
+		}
+		dst[k] = v
+	}
+	return dst
+}
+
+func lowerKeys(m map[string]any) map[string]any {
+	res := make(map[string]any, len(m))
+	for k, v := range m {
+		lk := strings.ToLower(k)
+		switch val := v.(type) {
+		case map[string]any:
+			res[lk] = lowerKeys(val)
+		case []any:
+			res[lk] = lowerKeysInSlice(val)
+		default:
+			res[lk] = val
+		}
+	}
+	return res
+}
+
+func lowerKeysInSlice(s []any) []any {
```

Comment:
What is the use case for lowercasing nested keys? I think maybe just lowercasing the variable name is all that's expected.